src/test/rgw/test_store_bench.cc

```diff
@@ -0,0 +1,345 @@
+              << std::endl;
+  }
+  std::cout << "create user: " << err << std::endl;
+}
+
+void list_users() {
+  RGWUserAdminOpState op_state(CONF.store);
+  JSONFormatter jf;
+  std::stringstream ss;
+  RGWStreamFlusher flusher(&jf, ss);
+  int err = RGWUserAdminOp_User::list(CONF.dpp, CONF.store, op_state, flusher);
+  std::cout << "list user: " << err << std::endl;
+  std::cout << "list " << ss.str() << std::endl;
+}
+
+void create_bucket() {
+  Config c = Config::instance();
+  RGWBucketInfo info;
+  RGWUserInfo owner;
+  obj_version objv;
+  rgw_placement_rule rule(
+      c.store->get_zone()->get_zonegroup().get_default_placement_name(),
+      RGW_STORAGE_CLASS_STANDARD);
+  std::map<std::string, bufferlist> attrs;
+  owner.user_id.id = Config::instance().uid.id;
+
+  // idk what this is
+  objv.ver = 2;
+  objv.tag = "write_tag";
+  rule.name = "rule1";
+  rule.storage_class = "sc1";
+  auto user = c.store->get_user(Config::instance().uid);
+  rgw_placement_rule r(
+      c.store->get_zone()->get_zonegroup().get_default_placement_name(),
+      RGW_STORAGE_CLASS_STANDARD);
+  owner.default_placement = r;
+  RGWQuotaInfo qi;
+  auto zone = c.store->get_zone();
+  RGWAccessControlPolicy cp;
+  RGWEnv env;
+  req_info req_i(c.cct, &env);
+  std::unique_ptr<rgw::sal::Bucket> bucket;
+  std::string swift_var_location;
+  bool existed = false;
+  int ret =
+      user->create_bucket(c.dpp, c.bucket, zone->get_zonegroup().get_id(), r,
+                          swift_var_location, &qi, cp, attrs, info, objv, false,
+                          false, &existed, req_i, &bucket, null_yield);
+  std::cout << "create bucket: " << ret << std::endl;
+  std::cout << "existed: " << existed << std::endl;
+}
+
+void list_buckets() {
+  auto c = Config::instance();
+  auto user = c.store->get_user(c.uid);
+  rgw::sal::BucketList buckets;
+  std::string marker, end_marker;
+  user->list_buckets(c.dpp, marker, end_marker, 10, 0, buckets, null_yield);
+  std::cout << "Buckets: ";
+  for (auto &[name, bucket] : buckets.get_buckets()) {
+    std::cout << name << " ";
+  }
+  std::cout << std::endl;
+}
+
+static bufferlist *bl = nullptr;
+// bl->length() doesn't work?????
+static uint32_t bufferlist_length = 0;
+
+uint64_t create_object(rgw_obj_key key, uint32_t obj_size) {
+  // For now let's have one bufferlist as append might be too expensive
+  if (bl == nullptr) {
+    bl = new bufferlist();
+    bl->append_zero(obj_size);
+    bufferlist_length = obj_size;
+  }
+  if (bufferlist_length != obj_size) {
+    bl->clear();
+    bl->append_zero(obj_size);
+    bufferlist_length = obj_size;
+  }
+
+  std::unique_ptr<rgw::sal::Object> obj = CONF.store->get_object(key);
+  std::unique_ptr<rgw::sal::Bucket> bucket{nullptr};
+  auto ruser = CONF.store->get_user(CONF.uid);
+  CONF.store->get_bucket(CONF.dpp, ruser.get(), CONF.bucket, &bucket,
+                         null_yield);
+  obj->set_bucket(bucket.get());
```

Comment:
Oops, missed this last time.  You have a bucket, so you should use bucket->get_object(), rather than store->get_bucket().  Then you don't need to set the bucket.

Reply:
done.